superset/assets/javascripts/explore/stores/controls.jsx

```diff
@@ -334,11 +334,14 @@ export const controls = {
     type: 'SelectControl',
     multi: true,
     label: 'Columns',
-    mapStateToProps: state => ({
-      choices: (state.datasource) ? state.datasource.gb_cols : [],
-    }),
     default: [],
     description: 'One or many controls to pivot as columns',
+    optionRenderer: c => <ColumnOption column={c} />,
```

Comment:
This file (`controls.jsx`) holds configuration for a "pool" of controls that can be used across visualizations. When creating a new visualization, you'll want to reuse controls that already exist. If the exact control that you need doesn't exist already, you'll want to create it here.

Note that `type` references controls React components defined in `javascripts/explore/component/controls/`.

Also note that controls can be dynamic with `mapStateToProps`, allowing to map anything from the app's state into a prop for the component. This allows in this case to get configuration elements proper to the datasource in this case, but can also be used to change a control based on another control's value.